### PR TITLE
filtering types

### PR DESCRIPTION
**Pull request by seb231**
The property types for the result of Google Places API call contains deprecated data that need to be removed. 
The ids from Google Places API are also unwanted.

### CODE REVIEW

File: data/unwanted_property_types.edn [hunk 1]
```diff
@@ -0,0 +1,40 @@
+["place_of_worship"
+"need to remove all type 2 google types"
```
Comment (acron0) [inline data/unwanted_property_types.edn]: really?

File: src/witan/gwyn/gwyn.clj [hunk 2]
```diff
@@ -67,10 +70,26 @@
...
 (defn collect-property [json-entry]
   (hash-map :address (get json-entry "vicinity")
-            :type (get json-entry "types")
+            :type (set (get json-entry "types"))
             :id (get json-entry "id")
             :name (get json-entry "name")))
 
+(def unwanted-properties
+  (edn/read-string
+   (slurp (io/file "data/unwanted_property_types.edn"))))
```
Comment (acron0) [inline src/witan/gwyn/gwyn.clj]: Why a file and not a resource, an input or just data in code? When JAR'd, models won't have access to external files

File: src/witan/gwyn/schemas.clj [hunk 3]
```diff
@@ -30,14 +30,15 @@
                            [:num-pumps-attending s/Int]]))
 
 (def HistoricalFireRiskScores
-  {})
+  (make-ordered-ds-schema [[:address s/Str] [:name s/Str] [:risk-score java.lang.Double]
+                           [:date-last-risk-assessed s/Any]]))
```
Comment (acron0) [inline src/witan/gwyn/schemas.clj]: Any chance of a schema for the date?

File: src/witan/gwyn/schemas.clj [hunk 4]
```diff
@@ -50,13 +51,16 @@
                            [:generic-fire-risk-score s/Int]]))
 
 (def CommercialPropertiesWithScores
-  {})
+  (make-ordered-ds-schema [[:address s/Str] [:name s/Str] [:risk-score java.lang.Double]
+                           [:date-last-risk-assessed s/Any]]))
```
Comment (acron0) [inline src/witan/gwyn/schemas.clj]: Any chance of a schema for the date?

File: src/witan/gwyn/schemas.clj [hunk 5]
```diff
@@ -50,13 +51,16 @@
...
 
 (def HistoricalFireRiskScores
-  {})
+  (make-ordered-ds-schema [[:address s/Str] [:name s/Str] [:risk-score java.lang.Double]
+                           [:date-last-risk-assessed s/Any]]))
```
Comment (acron0) [inline src/witan/gwyn/schemas.clj]: Any chance of a schema for the date?

File: src/witan/gwyn/schemas.clj [hunk 6]
```diff
@@ -50,13 +51,16 @@
...
 
 (def NewFireRiskScores
-  {})
+  (make-ordered-ds-schema [[:address s/Str] [:name s/Str] [:risk-score java.lang.Double]
+                           [:date-last-risk-assessed s/Any]]))
```
Comment (acron0) [inline src/witan/gwyn/schemas.clj]: Any chance of a schema for the date?

File: data/unwanted_property_types.edn (hunk 1 above)
Comment (Eleonore9) [inline data/unwanted_property_types.edn]: oops :confused: I copied a whole list (obviously). Will remove that!

File: src/witan/gwyn/gwyn.clj (hunk 2 above)
Comment (Eleonore9) [inline src/witan/gwyn/gwyn.clj]: I thought it'd be cleaner in an external file, but I can just add it to the utils ns and import it.

File: src/witan/gwyn/utils.clj [hunk 7]
```diff
@@ -11,3 +11,43 @@
   (if (zero? dd)
     0.0
     (/ ^double d dd)))
+
+(def unwanted-property-types
```
Comment (acron0) [inline src/witan/gwyn/utils.clj]: Sorry to be a pain but I'd probably put these into `gwyn.clj` as they're definitely contextually relevant, where as 'utils' is usually for stuff that is irrelevant or generic.

File: src/witan/gwyn/schemas.clj [hunk 8]
```diff
@@ -31,7 +32,7 @@
 
 (def HistoricalFireRiskScores
   (make-ordered-ds-schema [[:address s/Str] [:name s/Str] [:risk-score java.lang.Double]
-                           [:date-last-risk-assessed s/Any]]))
+                           [:date-last-risk-assessed sc/ISO-Date-Time]]))
```
Comment (seb231) [inline src/witan/gwyn/schemas.clj]: Are we confident about the human readability of ISO dates? We definitely don't need times. Dates will be input by hand by users into a csv

Comment (Eleonore9) [inline src/witan/gwyn/schemas.clj]: @seb231 I thought when we run the model we'd add a date to say this property was scored that much at that date.

**Comment by seb231** [comment]
@Eleonore9 I think there is a little confusion regarding the dates. The key is "date-last-risk-assessed", not date the property was last scored. Essentially all this is at this stage is a blank field that we want to output. When we get to the stage where historical risk assessments have been made (which affect the score), this will need to be reflected in the schema. Do we need to have a meeting around this @Eleonore9 @acron0 ?

File: src/witan/gwyn/gwyn.clj [hunk 9]
```diff
@@ -67,10 +71,36 @@
 
 (defn collect-property [json-entry]
   (hash-map :address (get json-entry "vicinity")
-            :type (get json-entry "types")
+            :type (set (get json-entry "types"))
             :id (get json-entry "id")
             :name (get json-entry "name")))
 
+(def unwanted-property-types
+  ["administrative_area_level_1" "administrative_area_level_2"
```
Comment (acron0) [inline src/witan/gwyn/gwyn.clj]: This can be a set

**Pull request merged**